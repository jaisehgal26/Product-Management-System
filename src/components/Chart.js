import React from "react";
// eslint-disable-next-line no-unused-vars
import Chart from "chart.js/auto";
import { useState, useEffect } from "react";
import axios from "axios";
import { Bar } from "react-chartjs-2";
import Fade from "react-reveal/Fade";

function GraphChart() {
  const [product, setProduct] = useState([]);
  const [price, setPrice] = useState([]);

  const selectChart = (e) => {
    axios.get(`http://localhost:3000/products`).then((res) => {
      const productData = res.data;
      let productName = [];
      let price = [];
      productData.forEach((element) => {
        productName.push(element.productname);
        price.push(element.price);
      });
      setProduct(productName);
      setPrice(price);
    });
  };
  useEffect(() => {
    selectChart();
  }, []);
  const data = {
    labels: product,
    datasets: [
      {
        label: "Products Price in $",
        data: price,
        fill: true,
        backgroundColor: "rgba(246, 211, 101, 1)",
        borderColor: "rgba(253, 160, 133, 1)",
      },
    ],
  };
  return (
    <Fade bottom>
      <hr />
      <Bar data={data} />
      <h6 className="text-center mt-1 Playfair">Products Pricing in USD$</h6>
      <hr />
    </Fade>
  );
}
export default GraphChart;
