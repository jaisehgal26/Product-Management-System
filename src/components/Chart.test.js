import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Chart from "./Chart";

test("renders Banner caption", () => {
  const wrapper = shallow(<Chart />);
  expect(wrapper.find("h6").text()).toContain("Products Pricing in USD$");
});
it("includes one h6", () => {
  const wrapper = shallow(<Chart />);
  expect(wrapper.find("h6").length).toEqual(1);
});
test("should render correctly", () => {
  const wrapper = toJson(shallow(<Chart />));
  expect(wrapper).toMatchSnapshot();
});
