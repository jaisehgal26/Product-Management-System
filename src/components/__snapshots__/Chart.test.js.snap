// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fade
  bottom={true}
>
  <hr />
  <ForwardRef
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(246, 211, 101, 1)",
            "borderColor": "rgba(253, 160, 133, 1)",
            "data": Array [],
            "fill": true,
            "label": "Products Price in $",
          },
        ],
        "labels": Array [],
      }
    }
  />
  <h6
    className="text-center mt-1 Playfair"
  >
    Products Pricing in USD$
  </h6>
  <hr />
</Fade>
`;
